mypy/dmypy_server.py

```diff
@@ -196,6 +196,7 @@ def serve(self) -> None:
 
     def free_global_state(self) -> None:
         TypeState.reset_all_subtype_caches()
+        TypeState.reset_protocol_deps()
```

Comment:
These who functions could be put in a function somewhere, since the two lines are repeated in `mypy.build` -- something like `reset_global_state`. It could be module-level function in mypy.typestate, and if we ever have multiple modules with global state, we can just move the function to another module.

Reply:
OK, I will make it a global function in `typestate`.